### PR TITLE
[Mime] Update mime types

### PR DESCRIPTION
**Pull request by smnandre**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      |  no
| New feature?  | no
| Deprecations? | no
| Issues        | Fix 
| License       | MIT

I had to update the prefered extensions map in `src/Symfony/Component/Mime/Resources/bin/update_mime_types.php`

```diff
-   'yaml' => ['application/yaml'],
-   'yml' => ['application/yaml'],
+   'yml' => ['application/yaml'],
+   'yaml' => ['application/yaml'],
```

This is the only way to make
* 'application/yaml' first type for 'yaml'  extension
* 'application/yaml' first type for 'yml' extension
* 'yaml' **first** extension for 'application/yaml' type
* 'yml' second extension for 'application/yaml' type

---

If it's still possible before 7.2 ?

(last one was 8 months ago)

### CODE REVIEW

File: src/Symfony/Component/Mime/MimeTypes.php [hunk 1]
```diff
@@ -2770,7 +2830,7 @@ public function guessMimeType(string $path): ?string
         'mp21' => ['application/mp21'],
         'mp2a' => ['audio/mpeg'],
         'mp3' => ['audio/mpeg', 'audio/mp3', 'audio/x-mp3', 'audio/x-mpeg', 'audio/x-mpg'],
-        'mp4' => ['video/mp4', 'video/mp4v-es', 'video/x-m4v'],
+        'mp4' => ['application/mp4', 'video/mp4', 'video/mp4v-es', 'video/x-m4v'],
```
Comment (fabpot) [inline src/Symfony/Component/Mime/MimeTypes.php]: This one and the next one seem suspicious to me.

Comment (smnandre) [reply src/Symfony/Component/Mime/MimeTypes.php]: application/mp4 is a registered mime type

* https://www.iana.org/assignments/media-types/application/mp4
* https://datatracker.ietf.org/doc/rfc4337/

Do you think we should add a special case for it to grant the first place to "video/mp4" (as we do for mp3 and audi) ?

https://github.com/symfony/symfony/blob/fa0a13fe0e450aba8def0b8a8362bb65ca8fc429/src/Symfony/Component/Mime/Resources/bin/update_mime_types.php#L92

**Comment by smnandre** [comment]
Updated preferences for aac (same thing), mp4 and mpeg4

Yaml has changed due to our preferences (not sure why it did not in first commit, and a small new change for "vor" in the meanwhile)

File: src/Symfony/Component/Mime/MimeTypes.php (hunk 1 above)
Comment (smnandre) [reply src/Symfony/Component/Mime/MimeTypes.php]: Done it / cf last comment

File: src/Symfony/Component/Mime/MimeTypes.php [hunk 2]
```diff
@@ -1249,7 +1278,7 @@ public function guessMimeType(string $path): ?string
         'application/xslt+xml' => ['xsl', 'xslt'],
         'application/xspf+xml' => ['xspf'],
         'application/xv+xml' => ['mxml', 'xhvml', 'xvml', 'xvm'],
-        'application/yaml' => ['yaml', 'yml'],
+        'application/yaml' => ['yml', 'yaml'],
```
Comment (fabpot) [inline src/Symfony/Component/Mime/MimeTypes.php]: Let's not change this one as this is commonly used.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Mime/MimeTypes.php (hunk 2 above)
Comment (smnandre) [reply src/Symfony/Component/Mime/MimeTypes.php]: Ok i understand where the problem is coming from.

I updated the issue comment and added  a comment in the update_mime_types.php file

(and fixed the MimeTypes.php file)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @smnandre.

**Pull request merged**